pkg/rtc/mediatrack.go

```diff
@@ -436,6 +436,8 @@ func (t *MediaTrack) SetMuted(muted bool) {
 
 func (t *MediaTrack) handleReceiverEverAddDowntrack() {
 	if !t.everSubscribed.Swap(true) && t.params.OnTrackEverSubscribed != nil {
-		go t.params.OnTrackEverSubscribed(t.ID())
+		go func() {
+			t.params.OnTrackEverSubscribed(t.ID())
```

Comment:
The `t.ID()` here was where things were deadlocking. The path was
- SetupReceiver grabbing the lock
- Calling DummyReceiver.Upgrade which does AddDownTrack on receiver
- That can trigger this callback

Doing this for now to prevent deadlock. Have changed DummyReceiver to not call `AddDownTrack` inside its lock, but will also look if it is possible to change `SetUpReceiver` to not hold the lock in the upgrade path.